### DISCUSSION TITLE
Error saving a demosaiced image: "Cannot handle this data type: (1, 1, 3), <f4"

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [colour-science/colour-demosaicing#58] [Q&A] [answered]
I'm trying to demosaic a BMP image based on "GRBG" Bayer pattern, this way:

```
img = colour.io.read_image("input.bmp")
coloured = demosaicing_CFA_Bayer_bilinear(img, "GRBG")
encoded = colour.cctf_encoding(coloured)
colour.io.write_image(encoded, "result.bmp")
```
But `write_image` throws `Cannot handle this data type: (1, 1, 3), <f4`

To be noted, the program shows this warning at the beginning: `ColourUsageWarning: "OpenImageIO" related API features are not available, switching to "Imageio"!`

If useful, this is the input image: [input.zip](https://github.com/colour-science/colour-demosaicing/files/13177705/input.zip)

Can you help me? Thanks!

**Comment by alessandrofrancesconi** [comment]
After installing OpenImageIO, it worked. Anyway, I'm wondering what could be the issue with Imageio.

**Reply by thomasmansencal** [reply, marked as answer]
Hi @alessandrofrancesconi,

We always convert the image type to `float32` and normalise it between [0, 1] if it is of an integer type:

```
>>> colour.io.read_image_Imageio("/Users/kelsolaar/Downloads/input.bmp").dtype
dtype('float32')
```

When it come to writing, *Imageio* is not as smart as *OpenImageIO* and tries to write that `float32` data into a 8-bit Bitmap container, so it fails.

Hope that makes sense!

Cheers,

Thomas
